File: Algorithm/Sorting/Insertion Sort/insertionSort.c
#include<stdio.h>
#include "../array.h"

void insertionSort(int* arr, int n){
    int i, key, j;
    for (int i = 1; i < n; i++)
    {
        key = arr[i];
        j = i - 1;

        while(j >= 0 && arr[j] > key){
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

int main()
{
    int arr[] = {9, 7, 2, 3, 1, 6, 7};
    int n = sizeof(arr)/sizeof(int);
    insertionSort(arr, n);
    printArray(arr, n);
    return 0;
}

File: Algorithms/Sorting/Insertion Sort/insertion_sort.py
def insertion_sort(arr):
    size = len(arr)
    for i in range(1, size):
        key = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1

        arr[j + 1] = key

arr = [4, 2, 5, 1, 7, 8]
insertion_sort(arr)
print(arr)
